tests/base/hdf5_05.cc

```diff
@@ -0,0 +1,1463 @@
+
+        pcout << "IO mode " + dataset_name << "<" << type_name << ">"
+              << " (Write): " << dataset.get_io_mode() << std::endl;
+        pcout << "Local no collective cause " + dataset_name << "<" << type_name
+              << ">"
+              << " (Write): " << dataset.get_local_no_collective_cause()
+              << std::endl;
+        pcout << "Global no collective cause " + dataset_name << "<"
+              << type_name << ">"
+              << " (Write): " << dataset.get_global_no_collective_cause()
+              << std::endl;
+      }
+    else
+      {
+        dataset.template write_none<Number>();
+        pcout << "IO mode " + dataset_name << "<" << type_name << ">"
+              << " (Write): " << dataset.get_io_mode() << std::endl;
+        pcout << "Local no collective cause " + dataset_name << "<" << type_name
+              << ">"
+              << " (Write): " << dataset.get_local_no_collective_cause()
+              << std::endl;
+        pcout << "Global no collective cause " + dataset_name << "<"
+              << type_name << ">"
+              << " (Write): " << dataset.get_global_no_collective_cause()
+              << std::endl;
+      }
+  }
+
+  {
+    std::string dataset_name("dataset_5");
+
+    // This dataset is a 3x2x4 tensor
+    // Output generated by h5dump
+    //  (0,0,0): 2, 0, 0, 38,
+    //  (0,1,0): 6, 5, 0, 3,
+    //  (1,0,0): 12, 0, 0, 0,
+    //  (1,1,0): 16, 15, 0, 13,
+    //  (2,0,0): 32, 0, 0, 0,
+    //  (2,1,0): 36, 35, 0, 33
+
+
+    const std::vector<hsize_t> dataset_dimensions = {3, 2, 4};
+    auto                       dataset =
+      group.create_dataset<Number>(dataset_name, dataset_dimensions);
+    dataset.set_query_io_mode(true);
+    deallog << "Dimensions " << dataset_name << "<" << type_name << ">"
+            << " (Write): " << dataset.get_dimensions() << std::endl;
+    deallog << "Size " << dataset_name << "<" << type_name << ">"
+            << " (Write): " << dataset.get_size() << std::endl;
+    deallog << "Rank " << dataset_name << "<" << type_name << ">"
+            << " (Write): " << dataset.get_rank() << std::endl;
+
+    if (Utilities::MPI::this_mpi_process(mpi_communicator) == 0)
+      {
+        std::vector<hsize_t> coordinates = {0,
+                                            0,
+                                            0, // first point
+                                            0,
+                                            1,
+                                            3, // second point
+                                            0,
+                                            1,
+                                            1, // third point
+                                            0,
+                                            1,
+                                            0}; // fourth point
+        std::vector<Number>  data        = {2, 3, 5, 6};
+
+        dataset.write_selection(data, coordinates);
+
```

Comment:
Write an unordered selection of points in a rank 3 dataset. This is probably the most flexible method and quite useful in the context of deal.ii. In an MPI simulation, the space is divided in areas with irregular shapes, therefore the data can not be separated in nice hyperslabs.